src/network_analysis.jl

```diff
@@ -192,6 +192,122 @@ function complexstoichmat(::Type{Matrix{Int}}, rn::ReactionSystem, rcs)
     Z
 end
 
+@doc raw"""
+    laplacianmat(rn::ReactionSystem, pmap=Dict(); sparse=false)
+
+    Return the negative of the graph Laplacian of the reaction network. The ODE system of a chemical reaction network can be factorized as ``\frac{dx}{dt} = Y A_k Φ(x)``, where ``Y`` is the [`complexstoichmat`](@ref) and ``A_k`` is the negative of the graph Laplacian, and ``Φ`` is the [`massactionvector`](@ref). ``A_k`` is an n-by-n matrix, where n is the number of complexes, where ``A_{ij} = k_{ij}`` if a reaction exists between the two complexes and 0 otherwise. 
+    Returns a symbolic matrix by default, but will return a numerical matrix if parameter values are specified via pmap. 
+"""
+function laplacianmat(rn::ReactionSystem, pmap = Dict(); sparse = false) 
+    D = incidencemat(rn; sparse); K = fluxmat(rn, pmap; sparse)
+    D*K
+end
+
+@doc raw"""
+    fluxmat(rn::ReactionSystem, pmap = Dict(); sparse=false)
+
+    Return an r×c matrix ``K`` such that, if complex ``j`` is the substrate complex of reaction ``i``, then ``K_{ij} = k``, the rate constant for this reaction. Mostly a helper function for the network Laplacian, [`laplacianmat`](@ref). Has the useful property that ``\frac{dx}{dt} = S*K*Φ(x)``, where S is the [`netstoichmat`](@ref) or net stoichiometry matrix and ``Φ(x)`` is the [`massactionvector`](@ref).
+    Returns a symbolic matrix by default, but will return a numerical matrix if rate constants are specified as a `Tuple`, `Vector`, or `Dict` of symbol-value pairs via `pmap`.
+"""
+function fluxmat(rn::ReactionSystem, pmap::Dict = Dict(); sparse=false) 
+    rates = reactionrates(rn)
+
+    !isempty(pmap) && (rates = substitutevals(rn, pmap, parameters(rn), rates))
+    rcmap = reactioncomplexmap(rn); nc = length(rcmap); nr = length(rates)
+    mtype = eltype(rates) <: Symbolics.BasicSymbolic ? Num : eltype(rates)
```

Comment:
I think it has to, doesn't seem like it's actually possible to have zeros in this matrix if the eltype is `BasicSymbolic`

Reply:
Can't it by type `Any`?

Reply:
Is that better than being `Num`? I assumed more specific was better

Reply:
We don't want to return mixtures of Nums and non-Nums across different methods, so we should not re-wrap internal symbolics.

Reply:
Could I just wrap it in an `Any` when I return it? It's useful to have it be `Num` at least temporarily because I use `zero(T)` and `one(T)` in various places (and product over an empty `Num[]` returns 1, which is useful).

Reply:
Actually this creates some issues for the sparse method (can't really do any arithmetic with a `SparseMatrixCSC{Any, T}` since zero(Any) is undefined). Would `Union{Int64, BasicSymbolic}` be okay? If not it might just be better to define two functions, one symbolic and one numeric

Reply:
If you want to internally use Num I guess that is ok, but yeah, we should unwrap before returning to users for consistency.

Reply:
What about returning a `Union` as the element type?

Reply:
Actually that doesn't work, nevermind, when you find the Laplacian matrix (D*K) type inference infers its eltype as Any anyway. Since it's the sparse matrix support that's causing the issue I think it might just be better to not allow the sparse kwarg for now

Reply:
Alright, I guess just stick with `Num`. But please add documentation about this and why it is needed to the function doctrings.